src/sentry/integrations/discord/integration.py

```diff
@@ -156,10 +158,6 @@ def build_integration(self, state: Mapping[str, object]) -> Mapping[str, object]
             },
         }
 
-    def setup(self) -> None:
```

Comment:
If this is removed then will self hosted users be able to install Discord? The base definition [here](https://github.com/getsentry/sentry/blob/master/src/sentry/integrations/base.py#L294-L300) says it should only be executed at runtime, so why is it happening more than once in production?

Also, it looks like the `discord.setup.update_bot_commands_failure` happens when there was an API error after attempting to update the commands - why weren't they found in the cache? Shouldn't they be?

Reply:
Good catch! I'm going to try changing this to post_install since this only needs to be registered once after install